src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Override
+    public Task<?> preDeactivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::remove);
+    }
+
+    private Task<?> doIfInfoContributor(final AbstractActor<?> actor,
+                                        final Consumer<ActorInfoContributorReference> synchronizedAction)
+    {
+        if (actor instanceof InfoContributor)
+        {
+            ActorInfoContributorReference actorInfoContributorReference = new ActorInfoContributorReference(actor);
+            synchronized (actorInfoContributorReferences)
+            {
+                synchronizedAction.accept(actorInfoContributorReference);
+            }
+        }
+        return Task.done();
+    }
+
+    @Override
+    public void contribute(final Info.Builder builder)
+    {
+        createActorInfoObject().ifPresent(actorInfoObject -> builder.withDetail("actors", actorInfoObject));
+    }
+
+    private Optional<?> createActorInfoObject()
+    {
+        Set<ActorInfoContributorReference> actorInfoContributorReferencesCopy;
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferencesCopy = actorInfoContributorReferences.stream()
+                    .map(ActorInfoContributorReference::new)
+                    .collect(Collectors.toSet());
+        }
+        Collection<ActorInfoContributorReference> expiredReferences = new ArrayList<>();
+        Map<String, Map<String, Object>> actorInfoObject = new HashMap<>();
+        actorInfoContributorReferencesCopy.forEach(actor ->
+        {
+            InfoContributor contributor = actor.reference.get();
+            if (contributor == null)
+            {
+                expiredReferences.add(actor);
+            }
+            else
+            {
+                final Info.Builder builder = new Info.Builder();
+                contributor.contribute(builder);
+                actorInfoObject.putIfAbsent(actor.name, new HashMap<>());
+                actorInfoObject.get(actor.name).put(actor.identity, builder.build().getDetails());
+            }
+        });
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferences.removeAll(expiredReferences);
+        }
+        if (actorInfoObject.isEmpty())
+        {
+            return Optional.empty();
+        }
+        else
+        {
+            return Optional.of(actorInfoObject);
+        }
+    }
+
+    private class ActorInfoContributorReference
+    {
+        private final WeakReference<InfoContributor> reference;
+        private final String name;
+        private final String identity;
+
+        private ActorInfoContributorReference(final AbstractActor<?> actor)
+        {
+            reference = new WeakReference<>((InfoContributor) actor);
+            name = orbitRemoteReferenceService.getInterfaceClass(actor).getSimpleName();
+            identity = ((Actor) actor).getIdentity();
```

Comment:
I might be totally wrong on this because I can't access the corpnet now, but I think MP does something funky with the actor's identity and I'm wondering if in pre-activation this returns the right identity for the account actor.